caches/infinispan/src/test/java/org/commonjava/maven/galley/cache/infinispan/FastLocalCacheProviderTest.java

```diff
@@ -15,138 +15,129 @@
-import org.commonjava.maven.galley.spi.cache.CacheProvider;
-import org.commonjava.maven.galley.spi.event.FileEventManager;
-import org.commonjava.maven.galley.spi.io.PathGenerator;
-import org.commonjava.maven.galley.spi.io.TransferDecorator;
-import org.infinispan.Cache;
-import org.infinispan.manager.EmbeddedCacheManager;
 import org.jboss.byteman.contrib.bmunit.BMScript;
 import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Ignore;
-import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import org.junit.rules.TestName;
 import org.junit.runner.RunWith;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
 import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.Executor;
-import java.util.concurrent.Executors;
 
 import static org.hamcrest.CoreMatchers.equalTo;
-import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertThat;
 
-/**
- */
 @RunWith( org.jboss.byteman.contrib.bmunit.BMUnitRunner.class )
-@BMUnitConfig( loadDirectory = "target/test-classes/bmunit", debug = true )
+@BMUnitConfig( loadDirectory = "target/test-classes/bmunit/common", debug = true )
 public class FastLocalCacheProviderTest
-        extends CacheProviderTCK
+        extends AbstractFastLocalCacheBMUnitTest
 {
-    private static EmbeddedCacheManager CACHE_MANAGER;
-
-    final String content = "This is a bmunit test";
-
-    final Location loc = new SimpleLocation( "http://foo.com" );
-
-    final String fname = "/path/to/my/file.txt";
-
-    final CountDownLatch latch = new CountDownLatch( 2 );
-
-    final ConcreteResource resource = new ConcreteResource( loc, fname );
-
-    private FastLocalCacheProvider provider;
-
-    private String result = null;
-
-    @Rule
-    public TemporaryFolder temp = new TemporaryFolder();
-
-    @Rule
-    public TestName name = new TestName();
-
-    private final PathGenerator pathgen = new HashedLocationPathGenerator();
-
-    private final FileEventManager events = new TestFileEventManager();
+    @Test
+    @BMScript( "LockThenWaitForUnLock.btm" )
+    public void testLockThenWaitForUnLock()
+            throws Exception
+    {
+        final Location loc = new SimpleLocation( "http://foo.com" );
+        final String path = "my/path.txt";
+        final ConcreteResource res = new ConcreteResource( loc, path );
 
-    private final TransferDecorator decorator = new TestTransferDecorator();
+        CountDownLatch latch = new CountDownLatch( 2 );
 
-    private Cache<String, String> cache = CACHE_MANAGER.getCache( NFSOwnerCacheProducer.CACHE_NAME );
+        new Thread( new WriteLockThread( res, latch ) ).start();
```

Comment:
I think I'd pull the thread init + start out into a method in the super-class, like this:

```
protected void start(Runnable r){
    new Thread(r).start();
}
```

This gives us a nice central way to make adjustments to the way all threads are started, like if we want to use an Executor, or use a thread naming scheme or something. We can also wrap the Runnable in something else for timing, if we want to...and it removes a little boilerplate code at the same time.

It's not that important, but it may save time later if we need to do that (later we'll probably have even more test cases).